hacking/update-sanity-requirements.py

```diff
@@ -45,7 +45,19 @@ def freeze_requirements(self) -> None:
             subprocess.run(pip + ['install', 'wheel'], env=env, check=True)  # make bdist_wheel available during pip install
             subprocess.run(pip + ['install', '-r', self.source_path], env=env, check=True)
 
-            pip_freeze = subprocess.run(pip + ['freeze'], env=env, check=True, capture_output=True, text=True)
+            keep_setuptools = any(line.startswith('setuptools ') for line in self.source_path.read_text().splitlines())
```

Comment:
The only thing I have to say here, is that this seems to rely on an implied means of listing requirements with a space following.  Dunno if it would be best to use some regex that smarter.

Reply:
The other option would be to run each line through `packaging.requirements.parse_requirement` and inspect the package name.

Reply:
Since the script is for our own use, and we control the input, I wasn't too worried about it not matching. It would only be a problem if we aren't paying attention when we update it, and then don't notice setuptools is dropped from the generated requirements file.

That being said, if it's something you're concerned about, I can put in something more robust.